src/components/@shared/DDODownloadButton/inputWithList.tsx

```diff
@@ -0,0 +1,79 @@
+import { getFieldContent } from '@utils/form'
+import content from '../../../../content/asset/form.json'
+import { Field, useFormikContext } from 'formik'
+import Input from '../FormInput'
+import Button from '../atoms/Button'
+import { useState } from 'react'
+import styles from './index.module.css'
+import { ListItem } from '../atoms/Lists'
+
+export default function InputWithList({
+  fieldname,
+  setList,
+  list
+}: {
+  fieldname: string
+  setList: (list: { id: string }[]) => void
+  list: { id: string }[]
+}) {
+  const [serviceCredential, setServiceCredential] = useState<string>()
+  const { setFieldValue, errors } = useFormikContext()
+  // add credential to the credentiallist
+  const addCredential = () => {
+    if (serviceCredential.trim() && !errors[fieldname]) {
+      setList([...list, { id: serviceCredential }])
+      setServiceCredential('')
+    }
+  }
+  // delete credential from the list
+  const deleteCredential = (index: number) => {
+    list.splice(index, 1)
+    setList([...list])
+  }
+  return (
+    <>
+      <div className={styles.credential_input_div}>
+        <div className={`${styles.credential_input}`}>
+          <Field
+            {...getFieldContent(fieldname, content.metadata.fields)}
+            value={serviceCredential}
+            onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+              setServiceCredential((e.target as HTMLInputElement).value)
+              setFieldValue(fieldname, (e.target as HTMLInputElement).value)
+              console.log(errors[fieldname])
```

Comment:
nitpick (blocking): Remove the console.log statement unless it’s meant for debugging purposes only. If necessary for debugging, use console.debug instead and ensure it’s removed or disabled in production.

Note: Also if you are logging errors consider using `console.error`.